test/higher_order_ops/test_invoke_subgraph.py

```diff
@@ -116,5 +118,95 @@ def fn(x):
         self.assertEqual(x.grad, x_clone.grad)
 
 
+@skipIfTorchDynamo("Not a torch._dynamo test")
+class TestInvokeSubgraphCompile(TestCase):
+    def test_simple(self):
+        def gn(x, y):
+            return (torch.mul(x, y),)
+
+        def fn(x, y):
+            return create_invoke_subgraph_op(gn, (x, y))[0]
+
+        x = torch.randn(8, requires_grad=True)
+        y = torch.randn(8, requires_grad=True)
+        ref = gn(x, y)[0]
+
+        x_clone = x.clone().detach().requires_grad_(True)
+        y_clone = y.clone().detach().requires_grad_(True)
+        res = torch.compile(fn, backend="eager", fullgraph=True)(x_clone, y_clone)
+
+        # Run backward
+        ref.sum().backward()
+        res.sum().backward()
+
+        self.assertEqual(ref, res)
+        self.assertEqual(x.grad, x_clone.grad)
+        self.assertEqual(y.grad, y_clone.grad)
+
+    def test_multiple(self):
+        def gn(x, y):
+            return (torch.mul(x, y),)
+
+        def fn(x, y):
+            a = create_invoke_subgraph_op(gn, (x, y))[0]
+            return create_invoke_subgraph_op(gn, (a, y))[0]
+
+        x = torch.randn(8, requires_grad=True)
+        y = torch.randn(8, requires_grad=True)
+        ref = fn(x, y)
+
+        x_clone = x.clone().detach().requires_grad_(True)
+        y_clone = y.clone().detach().requires_grad_(True)
+        backend = EagerAndRecordGraphs()
+        res = torch.compile(fn, backend=backend, fullgraph=True)(x_clone, y_clone)
+
+        # Run backward
+        ref.sum().backward()
+        res.sum().backward()
+
+        self.assertEqual(ref, res)
+        self.assertEqual(x.grad, x_clone.grad)
+        self.assertEqual(y.grad, y_clone.grad)
+
+        # Check that the Dynamo graph has just one subgraph module
+        self.assertEqual(len(backend.graphs), 1)
+        subgraph_attr_names = set()
+        for node in backend.graphs[0].graph.nodes:
+            if node.op == "get_attr":
+                subgraph_attr_names.add(node.target)
+        self.assertEqual(len(subgraph_attr_names), 1)
+
```

Comment:
Can we do an expecttest on the graphs? (I want to see them for one example, just to make sure things are hooked up correctly).